src/machines/test_help/async_workflow_driver.rs

```diff
@@ -138,6 +142,19 @@ impl CommandSender {
         }
     }
 
+    /// Request to run an activity
+    pub fn activity(&mut self, a: ScheduleActivity) -> impl Future {
+        let aid = a.activity_id.clone();
+        let c = WFCommand::AddActivity(a);
+        self.send(c);
+        let rx = self.twd_cache.add_sent_cmd(Activity(aid.clone()));
+        let cache_clone = self.twd_cache.clone();
+        async move {
+            cache_clone.set_cmd_blocked(Activity(aid));
+            rx.await
```

Comment:
This code and the `timer` code are almost totally the same except the actual map key part. Let's factor it out.